src/nhc2-platform.ts

```diff
@@ -87,34 +86,42 @@ class NHC2Platform implements DynamicPlatformPlugin {
     return this.accessories.find(accessory => accessory.UUID === device.Uuid);
   }
 
-  private addLights(accessories: Device[]) {
-    const lights = accessories.filter(light => light.Model === "light");
-    lights.forEach(light => {
-      const newAccessory = new Accessory(light.Name as string, light.Uuid);
-
-      const newService = new this.Service.Lightbulb(light.Name);
-      this.addStatusChangeCharacteristic(newService, newAccessory);
-      newAccessory.addService(newService);
-
-      this.processDeviceProperties(light, newService);
-
-      this.registerAccessory(newAccessory);
-    });
-  }
-
-  private addDimmers(accessories: Device[]) {
-    const dimmers = accessories.filter(light => light.Model === "dimmer");
-    dimmers.forEach(dimmer => {
-      const newAccessory = new Accessory(dimmer.Name as string, dimmer.Uuid);
-
-      const newService = new this.Service.Lightbulb(dimmer.Name);
-      this.addStatusChangeCharacteristic(newService, newAccessory);
-      this.addBrightnessChangeCharacteristic(newService, newAccessory);
-      newAccessory.addService(newService);
-
-      this.processDeviceProperties(dimmer, newService);
-
-      this.registerAccessory(newAccessory);
+  private addAccessories(accessories: Device[]) {
+    const mapping : { [index: string]: any } = {
+      "light" : {
+        "service" : this.Service.Lightbulb,
+        "handlers" : [
+          this.addStatusChangeCharacteristic
+        ]
+      },
+      "dimmer" : {
+        "service" : this.Service.Lightbulb,
+        "handlers" : [
+          this.addStatusChangeCharacteristic,
+          this.addBrightnessChangeCharacteristic
+        ]
+      },
+      "socket" : {
+        "service" : this.Service.Outlet,
+        "handlers" : [
+          this.addStatusChangeCharacteristic
+        ]
+      }
+    };
+
+    Object.keys(mapping).forEach(model => {
+      const config = mapping[model];
+      const accs = accessories.filter(acc => acc.Model === model);
+      accs.forEach(acc => {
+        const newAccessory = new Accessory(acc.Name as string, acc.Uuid);
+        const newService   = new config["service"](acc.Name);
```

Comment:
After removing the quotes from the dictionary you should be able to access the service through `config.service`